test: use a fork of ps-http-sim for security & stability

Jolg42 opened the pull request:
Before, we were doing a git clone on a repo that we don't own and its content could change at anytime.

Jolg42 committed:
test: use a fork of ps-http-sim for security & stability

Before we were doing a git clone on a repo that we don't own and its content could change at anytime.

miguelff commented on docker/planetscale_proxy/Dockerfile:
```diff
@@ -1,7 +1,7 @@
 
 RUN apt update && apt install netcat-openbsd -y
-RUN cd /go/src && git clone https://github.com/mattrobenolt/ps-http-sim.git
+RUN cd /go/src && git clone https://github.com/prisma/ps-http-sim.git
 RUN cd /go/src/ps-http-sim && go install .
 
 ENTRYPOINT /go/bin/ps-http-sim \
```
```suggestion
RUN cd /go/src && git clone https://github.com/prisma/planetscale-proxy.git
RUN cd /go/src/planetscale-proxy && go install .

ENTRYPOINT /go/bin/planetscale-proxy \
```

miguelff reviewed (approved):
Can you please 🙏🏾 use https://github.com/prisma/planetscale-proxy fork instead? I think the name is more descriptive. I provided a patch in the review.

Jolg42 committed:
Update docker/planetscale_proxy/Dockerfile

Co-authored-by: Miguel Fernández <fernandez@prisma.io>

Jolg42 commented:
Somehow it fails with the clone @miguelff, any idea? Do you see the same on your side?
```
container prisma-prisma-planetscale_proxy-1 is unhealthy
```
https://github.com/prisma/prisma/actions/runs/6481411961/job/17598870099?pr=21446#step:4:687

miguelff committed:
module name is preserved with go install

We could have use go build -i -o, but it doesn't matter actually. So let's go with the module name for the binary

janpio commented on docker/planetscale_proxy/Dockerfile:
```diff
@@ -1,8 +1,8 @@
+RUN cd /go/src && git clone https://github.com/prisma/planetscale-proxy.git
+RUN cd /go/src/planetscale-proxy && go install .
 
 ENTRYPOINT /go/bin/ps-http-sim \
```
might be worth a comment that the name comes from the codebase above (which now has no more connection in the name itself)

Jolg42 committed:
Update Dockerfile
